### PR TITLE
Chore: Replace backend platform codeownership

### PR DESCRIPTION
**Pull request by rhorvath**
**What is this feature?**
Code ownership cleanup, all references of Backend Platform are replaced.

**Special notes for your reviewer:**
I used backend group in the most of the cases. If you see a better match, please suggest an edit or directly add the team handle.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -58,106 +58,105 @@
...
+/pkg/services/navtree/ @grafana/grafana-backend-group
+/pkg/services/notifications/ @grafana/grafana-backend-group
+/pkg/services/org/ @grafana/grafana-backend-group
 /pkg/services/playlist/ @grafana/grafana-app-platform-squad
-/pkg/services/preference/ @grafana/backend-platform
-/pkg/services/provisioning/ @grafana/backend-platform
-/pkg/services/query/ @grafana/backend-platform
-/pkg/services/queryhistory/ @grafana/backend-platform
-/pkg/services/quota/ @grafana/backend-platform
-/pkg/services/rendering/ @grafana/backend-platform
```
Comment (hairyhenderson) [inline .github/CODEOWNERS:131]: this line seems to have been deleted (not just updated) - was that intended?

Comment (hairyhenderson) [reply .github/CODEOWNERS:131]: nm - added on ln586 to align more with the owning team

**Review by hairyhenderson — APPROVED** [review]
LGTM!

**Review by hairyhenderson — APPROVED** [review]

**Review by tolzhabayev — COMMENTED** [review]
what about the [go.mod ](https://github.com/grafana/grafana/blob/main/go.mod) file?

**Comment by rhorvath** [comment]
> what about the [go.mod ](https://github.com/grafana/grafana/blob/main/go.mod) file?

go.mod changes are included in this PR, am I missing something?

**Review by zserge — APPROVED** [review]

**Pull request merged**